app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -0,0 +1,757 @@
+    {
+        if ($request->hasFile('file')) {
+            $file = $request->file('file');
+            if ($file->getClientOriginalExtension() !== 'zip') {
+                return response()->json(['error' => 'Only ZIP files are allowed'], 400);
+            }
+            $directory = storage_path('app/public/shapefiles/' . uniqid('shapefile_'));
+            mkdir($directory, 0755, true);
+
+            // Extract the contents of the ZIP file
+            $zip = new \ZipArchive();
+            if ($zip->open($file->getPathname()) === true) {
+                $zip->extractTo($directory);
+                $zip->close();
+                $shpFile = $this->findShpFile($directory);
+                if (! $shpFile) {
+                    return response()->json(['error' => 'Shapefile (.shp) not found in the ZIP file'], 400);
+                }
+                $geojsonFilename = Str::replaceLast('.shp', '.geojson', basename($shpFile));
+                $geojsonPath = storage_path("app/public/geojson_files/{$geojsonFilename}");
+                $process = new Process(['ogr2ogr', '-f', 'GeoJSON', $geojsonPath, $shpFile]);
+                $process->run();
+                if (! $process->isSuccessful()) {
+                    Log::error('Error converting Shapefile to GeoJSON: ' . $process->getErrorOutput());
+
+                    return response()->json(['error' => 'Failed to convert Shapefile to GeoJSON', 'message' => $process->getErrorOutput()], 500);
+                }
+                $uuid = $this->insertGeojsonToDB($geojsonFilename);
+                if (isset($uuid['error'])) {
+                    return response()->json(['error' => 'Geometry not inserted into DB', 'message' => $uuid['error']], 500);
+                }
+
+                return response()->json(['message' => 'Shape file processed and inserted successfully', 'uuid' => $uuid], 200);
+            } else {
+                return response()->json(['error' => 'Failed to open the ZIP file'], 400);
+            }
+        } else {
+            return response()->json(['error' => 'No file uploaded'], 400);
+        }
+    }
+
+    private function findShpFile($directory)
+    {
+        $shpFile = null;
+        $files = scandir($directory);
+        foreach ($files as $file) {
+            if (pathinfo($file, PATHINFO_EXTENSION) === 'shp') {
+                $shpFile = "{$directory}/{$file}";
+
+                break;
+            }
+        }
+
+        return $shpFile;
+    }
+
+    public function checkSelfIntersection(Request $request)
+    {
+        $uuid = $request->query('uuid');
+        $geometry = PolygonGeometry::where('uuid', $uuid)->first();
+
+        if (! $geometry) {
+            return response()->json(['error' => 'Geometry not found'], 404);
+        }
+
+        $isSimple = DB::selectOne('SELECT ST_IsSimple(geom) AS is_simple FROM polygon_geometry WHERE uuid = :uuid', ['uuid' => $uuid])->is_simple;
+        $SELF_CRITERIA_ID = 4;
```

Comment:
What's the purpose of this magic number?

Reply:
I uploaded a table on the description of the PR, that is the number referenced to.